CMakeLists.txt

```diff
@@ -458,20 +458,30 @@ add_subdirectory(drivers)
 if(EXISTS ${CMAKE_BINARY_DIR}/zephyr_modules.txt)
 	file(STRINGS ${CMAKE_BINARY_DIR}/zephyr_modules.txt ZEPHYR_MODULES_TXT
 		ENCODING UTF-8)
+  set(module_names)
 
   foreach(module ${ZEPHYR_MODULES_TXT})
     # Match "<name>":"<path>" for each line of file, each corresponding to
     # one module. The use of quotes is required due to CMake not supporting
     # lazy regexes (it supports greedy only).
     string(REGEX REPLACE "\"(.*)\":\".*\"" "\\1" module_name ${module})
     string(REGEX REPLACE "\".*\":\"(.*)\"" "\\1" module_path ${module})
+
+    list(APPEND module_names ${module_name})
+
+    string(TOUPPER ${module_name} MODULE_NAME_UPPER)
+    set(ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR ${module_path})
```

Comment:
Just a naming bikeshed. Should we call this `ZEPHYR_MODULE_DIR_${MODULE_NAME_UPPER}` instead, so there is a consistent namespace (`ZEPHYR_MODULE_DIR_`) instead of "infixing" the module name like this?

Reply:
Thanks for the comment.
The reason for this naming is because it follows CMake style, such as:
https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html
`CMAKE_CURRENT_LIST_DIR`, `CMAKE_BINARY_DIR`, `CMAKE_CURRENT_SOURCE_DIR`, where `_DIR` is always final part.

This lead to the variable `ZEPHYR_CURRENT_MODULE_DIR` which were introduced earlier and follows existing CMake naming pattern.

Therefore I feel, `ZEPHYR_{NAME}_MODULE_DIR` follows nicely in this pattern, where `_CURRENT_` is replaced by the name of the module.